rotkehlchen/api/rest.py

```diff
@@ -3570,13 +3559,29 @@ def get_history_events(
                 cursor=cursor,
                 chain_id=chain_id,  # type: ignore
             )
+            ignored_ids_mapping = self.rotkehlchen.data.db.get_ignored_action_ids(
+                cursor=cursor,
+                action_type=ActionType.EVM_TRANSACTION,
+            )
 
+        # mypy fails to correctly detect the type here. It works fine now with pylance
```

Comment:
What is pylance? We use mypy in the codebase. Please don't add confusing comments. And you are adding a cast below. Ofc it will work with a cast. Just remove the comment if you want to keep the cast

Or what would make more sense would be to overload `get_all_history_events_and_limit_info` depending on the value of the boolean to return proper result.

But still not sure if it will work in this case as the value for `group_by_event_ids` is not a constant and I believe mypy would not follow the logic enough to see that the same argument is checked in the if above.

So I think I would keep the comment and add type ignores. So that if at some point mypy gets smarter and we add the overload we automatically get pinged that this is detected and fixed. Otherwise this cast would stay here forever

```suggestion
        # mypy fails to correctly detect the type here. It works fine now with pylance
```